packages/web3-utils/src/utils.js

```diff
@@ -231,6 +231,10 @@ var hexToNumber = function (value) {
         return value;
     }
 
+    if (_.isString(value) && !isHexStrict(value)) {
```

Comment:
This is probably a really bad idea (for hexToNumber). There are many places in the code where it's applied to values which might be decimal strings. Current behavior is to pass along to `toBN` without complaint..

Reply:
We should enforce to have hex values prefixed with ``0x`` or it should throw an error. Can you reference the code parts where a decimal string gets passed to ``hexToNumber``?

Reply:
You are right about this - I will look through and try to see if there is anything (because I'm still afraid :)

Reply:
@nivida Ok I looked and everything going to hexToNumber is output formatting so it should be fine. If you want to merge that's cool.

I would like to double-check this against the E2E PR before you publish a new version and add a test for events to make sure there's nothing weird coming from ganache.